internal/action/main_test.go

```diff
@@ -0,0 +1,291 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/stretchr/testify/assert"
+)
+
+var workspacePrefix string = "action-test"
+
+func NewTestRunConfig() *RunConfig {
+	return &RunConfig{
+		Token:                  os.Getenv("tf_token"),
+		Organization:           os.Getenv("tf_organization"),
+		Host:                   "app.terraform.io",
+		Name:                   fmt.Sprintf("%s-%d", workspacePrefix, time.Now().Unix()),
+		Import:                 true,
+		Apply:                  true,
+		TFEProviderVersion:     "0.25.3",
+		RunnerTerraformVersion: "1.0.5",
+		TerraformVersion:       "1.0.5",
+	}
+}
+
+func RemoveTestWorkspaces(ctx context.Context, client *tfe.Client, organization string, prefix string) error {
+	workspaces, err := client.Workspaces.List(ctx, organization, tfe.WorkspaceListOptions{
+		Search: &prefix,
+		ListOptions: tfe.ListOptions{
+			PageSize: 100,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, ws := range workspaces.Items {
+		if err := client.Workspaces.DeleteByID(ctx, ws.ID); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func TestCreateWorkspace(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+
+	config := NewTestRunConfig()
+
+	client, err := tfe.NewClient(&tfe.Config{
+		Address: fmt.Sprintf("https://%s", config.Host),
+		Token:   config.Token,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = client.Workspaces.Read(ctx, config.Organization, config.Name)
+	if err == nil {
+		t.Fatal("workspace should not exist, and an error should be returned")
+	}
+
+	if err.Error() != "resource not found" {
```

Comment:
Can you check this without string matching?

Reply:
will take a look, I think I took a look a while back and saw it was a plain error but I might have been mistaken

Reply:
https://github.com/hashicorp/go-tfe/blob/f5e4ef828c66335a7c6066e177bacc5cc99b9db1/tfe.go#L709-L749

So `errors.Is(err, tfe. ErrResourceNotFound)`

Reply:
You can check for error types or error values. 

You define a type when there's additional behavior, eg a `StatusCode` field that can be used when the error is unwrapped via `errors.As`. 

Error values are used a bunch in stdlib for well know errors and even occasionally termination conditions that aren't truly errors, ie `io.EOF`. It's almost like exporting a constant but errors can't be constants.